src/colmap/scene/image.h

```diff
@@ -161,14 +158,15 @@ class Image {
   camera_t camera_id_;
   struct Camera* camera_ptr_;
 
+  // The corresponding frame of the image. Note that multiple images might
+  // share the same frame. If not specified `kInvalidFrameId`.
+  frame_t frame_id_;
```

Comment:
Similarly as above, is storing the frame id necessary? This carries logic to keep it in sync with the frame ptr.